### PR TITLE
Apophenia 1.0 (new formula)

### PR DESCRIPTION
**Pull request by bdobyns**
### Have you:

- [X] Followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-science/blob/master/.github/CONTRIBUTING.md) document?
- [X] Checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-science/pulls) for the same update/change?
- [X] Run `brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting) and corrected all errors?
- [X] Built your formula locally prior to submission with `brew install <formula>`?

### New formula: have you

- [X] Written a sensible test? (The best test is to compile and run a sample program.)

### Other

- moved here to homebrew-science from homebrew-core https://github.com/Homebrew/homebrew-core/pull/331 at the suggestion of @tdsmith and @sjackman

### CODE REVIEW

File: apophenia.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+class Apophenia < Formula
+  desc "C library for statistical and scientific computing"
+  homepage "http://apophenia.info/"
+  url "https://github.com/b-k/apophenia/archive/v1.0.tar.gz"
+  sha256 "c753047a9230f9d9e105541f671c4961dc7998f4402972424e591404f33b82ca"
```
Comment (sjackman) [inline apophenia.rb]: Please add
```ruby
head "https://github.com/b-k/apophenia.git"
```

File: apophenia.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+  depends_on "gsl"
+
+  def install
+    system "./configure", "--with-mysql=no", "--enable-extended-tests", "--prefix=#{prefix}"
```
Comment (sjackman) [inline apophenia.rb]: Perhaps add
```ruby
depends_on "mysql" => :optional
```
and modify this line appropriately.

File: apophenia.rb (hunk 1 above)
Comment (bdobyns) [inline apophenia.rb]: for the very brave, sure.   building from head doesn't work for me.

File: apophenia.rb (hunk 2 above)
Comment (bdobyns) [inline apophenia.rb]: oh cool!  I couldn't figure out the syntax for that!  thank you!

**Comment by sjackman** [comment]
```
In file included from apop_dconstrain.c:1:0:
../apop_internal.h:38:21: fatal error: sqlite3.h: No such file or directory
```
https://travis-ci.org/Homebrew/homebrew-science/jobs/123967548#L409
`sqlite` is required on Linux. Please add
```ruby
depends_on "sqlite" unless OS.mac?
```
Do you have a Linux device on which to test?

File: apophenia.rb (hunk 1 above)
Comment (sjackman) [inline apophenia.rb]: What's the error message?

**Comment by sjackman** [comment]
Once it's ready, please squash the commits down to a single commit.
If you need help with that, see http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

File: apophenia.rb (hunk 1 above)
Comment (bdobyns) [inline apophenia.rb]: part of the problem is that the top of the repo as checked out is not build-able in the same way as the tarball as fetched from github.   you have to do something slightly different.

for the tarball, it's the usual `./configure && make && make install`

for the raw repo, it's `./configure && cd $somedir && make && make install`

**Comment by bdobyns** [comment]
@sjackman: I didn't realize homebrew ran anywhere other than osx, or needed to.  I can try to run homebrew homebrew on linux in a virtual-box

**Comment by sjackman** [comment]
No worries. Homebrew-core doesn't directly support Linux, but Homebrew-science does.

**Comment by bdobyns** [comment]
okay.   seems to build fine in the docker container sjackman/linuxbrew-standalone.    

I didn't know that was  possible.

**Comment by bdobyns** [comment]
darn.  Travis is failing.  

when I compare my failing Travis build https://travis-ci.org/Homebrew/homebrew-science/jobs/124013286  to a recent one that succeeded https://travis-ci.org/Homebrew/homebrew-science/jobs/123744911 I don't understand why it's failing. 

my build seems to fail in linuxbrew at the `brew tests` step, which seems like it should be well before doing anything particular to my formula.   that step works in my docker container.

**Comment by bdobyns** [comment]
I notice some formulae in science reference the DOI. which seems sensible.

I don't have a DOI for the book which this library accompanies, but I do have an 
* ISBN 978-0-691-13314-0
* eISBN: 978-1-4008-2874-6
* jstor stable url http://www.jstor.org/stable/j.ctt7t0dt
* url to a free copy on the author's page
  http://ben.klemens.org/pdfs/gsl_stats.pdf

**Comment by sjackman** [comment]
https://travis-ci.org/Homebrew/homebrew-science/jobs/124026271#L447-L461
```
==> brew audit homebrew/science/apophenia --strict --online
==> brew style apophenia
==> Installing or updating 'rubocop' gem
Successfully installed unicode-display_width-0.3.1
Successfully installed ruby-progressbar-1.7.5
Successfully installed rainbow-2.1.0
Successfully installed powerpack-0.1.1
Successfully installed ast-2.2.0
Successfully installed parser-2.3.0.7
Successfully installed rubocop-0.37.2
7 gems installed
1 file inspected, no offenses detected
Error: Broken pipe
==> FAILED
```

**Comment by sjackman** [comment]
This Travis failure is not caused by your formula. It's due to a bug in `brew test-bot`. So not your problem to worry about. Brew install and brew test worked just fine, so I think you're good to go.

**Comment by sjackman** [comment]
The `linuxbrew/linxubrew` image is more current than `sjackman/linuxbew*`.

**Comment by sjackman** [comment]
```
$ bundle --version
/Users/travis/build.sh: line 45: bundle: command not found
```
I don't see this error in the Travis build log. You can link to the exact line in the Travis build log by clicking on the line number.

**Comment by sjackman** [comment]
We don't have any ISBN in Homebrew-science formula, but I don't see any harm in adding it in as a comment.
```ruby
# isbn "978-0-691-13314-0"
```

**Comment by bdobyns** [comment]
here's the travis fail I referred to earlier
https://travis-ci.org/Homebrew/homebrew-science/jobs/124026270#L48

**Comment by sjackman** [comment]
I never noticed that error. I don't think it's a problem, but who knows. The `bundler` gem is later installed:
https://travis-ci.org/Homebrew/homebrew-science/jobs/124026270#L172

The final error:
`No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.`
https://travis-ci.org/Homebrew/homebrew-science/jobs/124026270#L201
I've seen frequently, but only for Mac OS. I only look at the Linux results of Travis and look at the BrewTestBot results for Mac OS.

**Comment by bdobyns** [comment]
when I run `brew tests` myself inside a docker container, it does take a while (although not 10m0s) to do whatever is going on

```
linuxbrew@e052da90a82b:~/.linuxbrew/Library/Taps/homebrew/homebrew-science$ brew tests
Run options: --seed 59803

# Running:

.................................................................................................................................................................................................................S...............................................S...S.........................................................................................................................SS..........................................................................................................................................................................................................................................................................................................................................................................................................................................................................

Finished in 78.354428s, 10.7588 runs/s, 25.8696 assertions/s.

843 runs, 2027 assertions, 0 failures, 0 errors, 5 skips

You have skipped tests. Run with --verbose for details.
linuxbrew@e052da90a82b:~/.linuxbrew/Library/Taps/homebrew/homebrew-science$
```

so whatever it is doing (and wherever it's doing it), it's seven times slower than my laptop.

**Comment by sjackman** [comment]
Merged. Thanks for your contribution, Barry!

**Comment by bdobyns** [comment]
Woo hoo!  All this so my daughter could `brew install apophenia`

Thank you for all your help!

**Comment by sjackman** [comment]
I hope that her gratitude is worth the effort. =)

**Comment by sjackman** [comment]
The `Broken pipe` error above may be caused by the following error:
```
14.00s==> brew audit homebrew/science/apophenia --strict --online
==> brew style apophenia
==> Installing or updating 'rubocop' gem
Successfully installed rainbow-2.1.0
Successfully installed ast-2.2.0
Successfully installed parser-2.3.0.7
Successfully installed powerpack-0.1.1
Successfully installed ruby-progressbar-1.7.5
Successfully installed unicode-display_width-1.0.3
Successfully installed rubocop-0.39.0
7 gems installed
1 file inspected, no offenses detected
Error: GitHub API Error: API rate limit exceeded for 208.78.110.199. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)
Try again in 41 minutes 58 seconds, or create a personal access token:
  https://github.com/settings/tokens/new?scopes=&description=Homebrew
and then set the token as: export HOMEBREW_GITHUB_API_TOKEN="your_new_token"
==> FAILED
```
https://travis-ci.org/Homebrew/homebrew-science/jobs/124051316#L282

Travis should have the environment variable `HOMEBREW_GITHUB_API_TOKEN` set.

**Pull request closed without merging**